ant/src/main/java/com/google/errorprone/ErrorProneExternalCompilerAdapter.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.tools.ant.types.Path;
+import org.apache.tools.ant.types.Reference;
+import org.apache.tools.ant.util.JavaEnvUtils;
+import org.apache.tools.ant.util.LoaderUtils;
+
+/**
+ * Ant component to launch an external javac with error-prone enabled.
+ */
+public class ErrorProneExternalCompilerAdapter extends DefaultCompilerAdapter {
+  private Path classpath;
+  private boolean suggestFixes;
+  private String memoryStackSize;
+  private List<Argument> jvmArgs = new ArrayList<Argument>();
+
+  public void setClasspath(Path classpath) {
+    this.classpath = classpath;
+  }
+
+  public Path createClasspath() {
+    if (classpath == null) {
+      classpath = new Path(getProject());
+    }
+
+    return classpath.createPath();
+  }
+
+  public void setClasspathRef(Reference r) {
+    createClasspath().setRefid(r);
+  }
+
+  public void setSuggestFixes(boolean suggestFixes) {
+    this.suggestFixes = suggestFixes;
+  }
+
+  public void setMemoryStackSize(String memoryStackSize) {
+    this.memoryStackSize = memoryStackSize;
+  }
+
+  public Argument createJvmArg() {
+    Argument arg = new Argument();
+    jvmArgs.add(arg);
+    return arg;
+  }
+
+  @Override
+  public boolean execute() throws BuildException {
+    if (getJavac().isForkedJavac()) {
+      attributes.log("Using external error-prone compiler", Project.MSG_VERBOSE);
+      Commandline cmd = new Commandline();
+      cmd.setExecutable(JavaEnvUtils.getJdkExecutable("java"));
+      if (memoryStackSize != null) {
+        cmd.createArgument().setValue("-Xss" + memoryStackSize);
+      }
+      String memoryParameterPrefix = "-X";
+      if (memoryInitialSize != null) {
+        cmd.createArgument().setValue(memoryParameterPrefix + "ms" + this.memoryInitialSize);
+        // Prevent setupModernJavacCommandlineSwitches() from doing it also
+        memoryInitialSize = null;
+      }
+      if (memoryMaximumSize != null) {
+        cmd.createArgument().setValue(memoryParameterPrefix + "mx" + this.memoryMaximumSize);
+        // Prevent setupModernJavacCommandlineSwitches() from doing it also
+        memoryMaximumSize = null;
+      }
+      for (Argument arg : jvmArgs) {
+        for (String part : arg.getParts()) {
+          cmd.createArgument().setValue(part);
+        }
+      }
+
+      cmd.createArgument().setValue("-classpath");
+      if (classpath == null) {
+        classpath = new Path(getProject());
+      }
+      // Usually redundant, but check two resources in case Ant stuff is in a different jar
+      addResourceSource(classpath, "com/google/errorprone/ErrorProneExternalCompilerAdapter.class");
+      addResourceSource(classpath, "com/google/errorprone/ErrorProneCompiler.class");
+      addResourceSource(classpath, "com/sun/tools/javac/Main.class");
+      cmd.createArgument().setPath(classpath);
+      cmd.createArgument().setValue(ErrorProneCompiler.class.getName());
+      if (suggestFixes) {
+        cmd.createArgument().setValue("-Xjcov");
```

Comment:
Did that change recently (in those last 88 changes)? When I initially created this it did seem to change the behavior, though it seemed a little unreliable.